### ISSUE TITLE
Add support to distinguish between diagnostics created by a reconciler and a builder

### ISSUE THREAD

**Issue by dbaeumer** [microsoft/vscode#74524]
Short summary: currently it is very hard to handle / merge diagnostics that come from to different source having the same owner. This either leads to stale problems or missing problems. For a lengthy discussion see https://github.com/Microsoft/vscode/issues/61140

Having support to distinguish between diagnostics create by a reconciler and a builder would help solve that problem.

**Comment by dbaeumer** [comment]
One idea would be two have two different tags that can be used. @jrieken & @mjbvz would that be in line with the current diagnostic design?

**Comment by mjbvz** [comment]
@dbaeumer Are you talking about the `DiagnosticTag`? I think that would make sense.

Or we could make it even more explicit by adding something like a `creator?: 'builder' | 'task'| ...` field on diagnostics

**Comment by dbaeumer** [comment]
@mjbvz yes I am talking about `DiagnosticTag`. The only disadvantage I see is that we need to search the array to determine whether a diagnostic is from a builder or a reconciler. Having a `creator` property would make this easier to distinguish.
